src/dataflow/src/source/s3.rs

```diff
@@ -934,24 +944,20 @@ impl SourceReader for S3SourceReader {
                     value: record,
                 }))
             }
-            Some(Some(Err(e))) => {
-                tracing::warn!(
-                    "when reading source '{}' ({}): {}",
-                    self.source_name,
-                    self.id,
-                    e
-                );
-                match e {
-                    S3Error::ClientConstructionFailed(err) => {
-                        Err(anyhow!("Client construction failed: {}", err))
-                    }
-                    S3Error::RetryFailed => Err(anyhow!("Retry failed")),
-                    S3Error::IoError(e) => Err(e.into()),
-                    S3Error::GetObjectError { .. } | S3Error::ListObjectsFailed(_) => {
-                        Ok(NextMessage::Pending)
-                    }
+            Some(Some(Err(e))) => match e {
+                S3Error::GetObjectError { .. } => {
+                    tracing::warn!(
+                        "when reading source '{}' ({}): {}",
```

Comment:
```suggestion
                        "error when reading source '{}' ({}): {}",
```

Reply:
I don't think that adding `error` to `warn!` logs is helpful, it just makes `grep error` (and especially case-insensitive search in e.g. `less`) less useful/more confusing. This will currently show up as `WARN <mod> when reading source...`.

I agree that this is a sentence fragment, though.